### PR TITLE
Faster install tests and updates

### PR DESCRIPTION
**Pull request by jhford**
Fix problems with tests.  There's two major changes here:

1. support session key for temporary aws credentials
2. use non-vhost addressing for tests.  We have substantial coverage for generating requests for the vhost case, but for macs this allows us to not need to add to the hosts file.

### CODE REVIEW

**Comment by jhford** [comment]
(I've verified this works locally, so if it fails on travis it's due to an absence of aws credentials)

**Comment by jhford** [comment]
@imbstack I'm asking for review even though travis isn't working.  I'm not sure how we put aws credentials into travis these days now that we have requiremfa.  I can verify that these tests work with a local AWS temporary credential.

**Comment by djmitche** [comment]
I created an IAM user and added its credentials to Travis.  What credentials was it using before?  Hopefully not your own! :)

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,11 @@
 language: node_js
-
 env:
-  - CXX=g++-4.8
-
+  global:
+    - CXX=g++-4.8
+    - SKIP_REAL_S3_TESTS=yes
```
Comment (djmitche) [inline .travis.yml]: You can remove this now that the creds are set up..

**Review by djmitche — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -1,54 +1,38 @@
...
     "uuid": "^3.3.2"
   },
   "scripts": {
-    "compile": "babel-compile -p taskcluster src:lib test-src:test",
-    "prepublish": "npm run compile",
-    "pretest": "npm run compile",
     "test": "DEBUG= mocha $MOCHA_FLAGS"
   },
+  "engines": {
+    "node": ">=10.0.0",
```
Comment (imbstack) [inline package.json]: Why 10? Does this use things not available in 8?

**Review by imbstack — APPROVED** [review]
Otherwise this looks fine to me! Also the notes that dustin brings up obviously.

File: package.json (hunk 2 above)
Comment (jhford) [reply package.json]: I've added 8 back, it shouldn't use anything 10 specific.

**Pull request merged**